src/internal.h

```diff
@@ -37,6 +37,10 @@ NS_INTERNAL struct ns_connection *ns_finish_connect(struct ns_connection *nc,
 NS_INTERNAL int ns_parse_address(const char *str, union socket_address *sa,
                                  int *proto, char *host, size_t host_len);
 
+#ifdef NS_ENABLE_COAP
```

Comment:
let's keep module specific API inside the module itself.
Stuff in internal is core. For example there are no MQTT things in `src/internal.h`